protocol/messenger.go

```diff
@@ -3208,10 +3208,20 @@ func (r *ReceivedMessageState) addNewActivityCenterNotification(publicKey ecdsa.
 		return fmt.Errorf("chat ID '%s' not present", message.LocalChatID)
 	}
 
+	// Use albumId as notificationId to prevent multiple notifications
+	// for same message with multiple images
+	var idToUse string
+	
+	if message.GetImage() != nil {
+		idToUse = message.GetImage().GetAlbumId()
```

Comment:
@ibrkhalil, I'm thinking now that the notification ID is linked to the album ID, what if all images in the message are removed and there is no album ID anymore? I believe this is possible, I'm just not totally sure.

Hey @borismelnik, do you know if what I mentioned above is a valid concern?

Reply:
@ilmotta I think usage `AlbumId` as notification ID is a good idea, cos the same concept we use for messages (MessageID). But need to process deletion as well. The same I did for deleted message, I think need to check it, if deleted, message was with AlbumId and was not last then need to skip sending empty notification :)

Reply:
@ibrkhalil but if last message in Album was deleted, need to send empty notification with correct id (in this context its an AlbumID)

Reply:
[there is my PR for deleting notification](https://github.com/status-im/status-go/pull/3277) may helps :)

Reply:
On Status-mobile, We disable editing messages with images. [ref](https://github.com/status-im/status-mobile/blob/47a8d20940f2f9d374207a22a57c75e9f16f95ba/src/status_im2/contexts/chat/messages/drawers/view.cljs#L31-L37)
So when we enable it later this might be an issue
Just tested Deletion, Deleting an image with mentions deletes the notification associated with the mention.

Reply:
@borismelnik, thanks for the insights. I guess then we should process deletion as you say to play on the safe side, even if now it works in mobile. Wdyt @ibrkhalil?

Reply:
Do you mean cascading the deletion from DB the notification deletion with the message deletion?

Reply:
@ibrkhalil I was worried that since the album ID depends on the existence of images, that if all images are removed, then the notification would have an ID, but one that doesn't refer to a valid album. We just need to guarantee that deleting the album would delete the notification I guess?

Reply:
> @ibrkhalil I was worried that since the album ID depends on the existence of images, that if all images are removed, then the notification would have an ID, but one that doesn't refer to a valid album. We just need to guarantee that deleting the album would delete the notification I guess?

Tested deleting the album and the notification vanished